### PR TITLE
Meetup management interface

### PR DESCRIPTION
**Pull request by asgrim**
First part of #105 

This PR allows creation of meetup data within the management console, as well as some style improvements.

I did *try* to keep it small, but there was still a lot of setting up to do.

*NOTE* Talks cannot be viewed/added/edited/removed yet, that will come in a follow-up PR to this, I didn't want it to grow even more.

### CODE REVIEW

File: test/AppTest/Action/Account/Meetup/EditMeetupActionTest.php [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+            $this->renderer,
+            $this->form,
+            $this->entityManager,
+            $this->findMeetup,
+            $this->findLocation,
+            $this->urlHelper
+        );
+    }
+
+    public function testGetRequestRendersTemplate()
```
Comment (lboynton) [inline test/AppTest/Action/Account/Meetup/EditMeetupActionTest.php]: What about an invalid GET request?

File: src/App/Entity/Meetup.php [hunk 2]
```diff
@@ -80,7 +79,6 @@ public static function fromStandardMeetup(
         DateTimeImmutable $from,
         DateTimeImmutable $to,
         Location $location,
-        array $talks,
         string $topic = null
     ) : self {
         $meetup = new self();
```
Comment (lboynton) [inline src/App/Entity/Meetup.php]: Do we also want to check that `toDate` is after `fromDate` here?

File: public/index.php [hunk 3]
```diff
@@ -1,6 +1,8 @@
 <?php
 declare(strict_types=1);
 
+Locale::setDefault('en_GB');
```
Comment (asgrim) [inline public/index.php]: done

File: config/autoload/dependencies.global.php [hunk 4]
```diff
@@ -14,7 +14,12 @@
             Application::class => ApplicationFactory::class,
             Helper\UrlHelper::class => Helper\UrlHelperFactory::class,
             App\Service\User\FindUserByEmailInterface::class => App\Service\User\FindUserByEmailFactory::class,
+            App\Service\Meetup\FindMeetupByUuidInterface::class => App\Service\Meetup\FindMeetupByUuidFactory::class,
             App\Service\Meetup\MeetupsServiceInterface::class => App\Service\Meetup\MeetupsServiceFactory::class,
+            App\Service\Meetup\GetAllMeetupsInterface::class => App\Service\Meetup\GetAllMeetupsFactory::class,
+            App\Service\Location\GetAllLocationsInterface::class => App\Service\Location\GetAllLocationsFactory::class,
+            App\Service\Location\FindLocationByUuid::class => App\Service\Location\FindLocationByUuidFactory::class,
```
Comment (asgrim) [inline config/autoload/dependencies.global.php]: done

File: src/App/Service/Location/GetAllLocationsInterface.php [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+declare(strict_types = 1);
+
+namespace App\Service\Location;
+
+use App\Entity\Location;
+
+interface GetAllLocationsInterface
+{
+    /**
+     * Return all locations. Returns an empty array if there none exist.
```
Comment (asgrim) [inline src/App/Service/Location/GetAllLocationsInterface.php]: done

File: src/App/Action/Account/Meetup/EditMeetupAction.php [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        $this->form = $form;
+        $this->entityManager = $entityManager;
+        $this->findLocationByUuid = $findLocationByUuid;
+        $this->urlHelper = $urlHelper;
+        $this->findMeetupByUuid = $findMeetupByUuid;
+    }
+
+    public function __invoke(Request $request, Response $response, callable $next = null) : Response
+    {
+        $meetup = $this->findMeetupByUuid->__invoke(Uuid::fromString($request->getAttribute('uuid')));
```
Comment (asgrim) [inline src/App/Action/Account/Meetup/EditMeetupAction.php]: Nah, it's not necessary really; it's fine to just let it bomb for now (also, it's admin-only)

Comment (asgrim) [inline src/App/Action/Account/Meetup/EditMeetupAction.php]: Yeah, that's possible, but annoyingly IDE support is shaky for that syntax (it doesn't error as such, type detection doesn't work) :/

File: src/App/Action/Account/Meetup/ViewMeetupAction.php [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        TemplateRendererInterface $templateRenderer,
+        FindMeetupByUuidInterface $findMeetupByUuid
+    ) {
+        $this->templateRenderer = $templateRenderer;
+        $this->findMeetupByUuid = $findMeetupByUuid;
+    }
+
+    public function __invoke(Request $request, Response $response, callable $next = null) : Response
+    {
+        $meetup = $this->findMeetupByUuid->__invoke(Uuid::fromString($request->getAttribute('uuid')));
```
Comment (asgrim) [inline src/App/Action/Account/Meetup/ViewMeetupAction.php]: No problemo :)

File: src/App/Service/Meetup/GetAllMeetupsInterface.php [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+declare(strict_types = 1);
+
+namespace App\Service\Meetup;
+
+use App\Entity\Meetup;
+
+interface GetAllMeetupsInterface
+{
+    /**
+     * Return all Meetup. Returns an empty array if there none exist.
```
Comment (asgrim) [inline src/App/Service/Meetup/GetAllMeetupsInterface.php]: Fixed

File: src/App/Service/Speaker/GetAllSpeakersInterface.php [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+declare(strict_types = 1);
+
+namespace App\Service\Speaker;
+
+use App\Entity\Speaker;
+
+interface GetAllSpeakersInterface
+{
+    /**
+     * Return all Speakers. Returns an empty array if there none exist.
```
Comment (asgrim) [inline src/App/Service/Speaker/GetAllSpeakersInterface.php]: Obvious was it? :D

File: test/AppTest/Action/Account/Meetup/EditMeetupActionTest.php (hunk 1 above)
Comment (asgrim) [inline test/AppTest/Action/Account/Meetup/EditMeetupActionTest.php]: It'll just throw an exception, but the exception is tested elsewhere; not necessary to retest in the action IMO

File: src/App/Entity/Meetup.php (hunk 2 above)
Comment (asgrim) [inline src/App/Entity/Meetup.php]: Good idea, added in

File: src/App/Action/Account/Meetup/EditMeetupAction.php (hunk 6 above)
Comment (lboynton) [inline src/App/Action/Account/Meetup/EditMeetupAction.php]: Ah yeah, I hadn't noticed that. That's a shame.

**Pull request merged**